### PR TITLE
Use Go 1.21.2

### PR DESCRIPTION
**Pull request by abitrolly**
With latest security fixes

https://go.dev/doc/devel/release#go1.21.minor

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ on:
     branches:
       - main
   pull_request:
+  workflow_dispatch:
```
Comment (keegancsmith) [inline .github/workflows/ci.yml:6]: why was this added?

**Review by keegancsmith — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml (hunk 1 above)
Comment (abitrolly) [reply .github/workflows/ci.yml:6]: To make it possible to run tests in branches in forks.
